lib/galaxy/config/__init__.py

```diff
@@ -229,11 +292,31 @@ def _create_attributes_from_raw_config(self):
                 raise ConfigurationError("Attempting to override existing attribute '%s'" % key)
             setattr(self, key, value)
 
-    def _process_config(self, kwargs):
-        self.config_dict = kwargs
-        self.root = find_root(kwargs)
-        self._set_config_base(kwargs)
+    def _resolve_paths(self, kwargs):
+
+        def resolve(key):
+            if key in _cache:  # resolve each path only once
+                return _cache[key]
+
+            path = getattr(self, key)  # path prior to being resolved
+            parent = self.appschema[key].get('path_resolves_to')
+            if not parent:  # base case: nothing else needs resolving
+                return path
+            parent_path = resolve(parent)  # recursively resolve parent path
+            if path is not None:
+                path = os.path.join(parent_path, path)  # resolve path
+            else:
+                path = parent_path  # or use parent path
```

Comment:
@ic4f Why override a `None` with the `parent_path` ?

Reply:
Because a path for an option can be not specified, while the `path_resolves_to` attribute *will* be.  In which case, we want to fall back to the parent path of that option. For example:

optionA: has a value 'myfile', resolves to: 'optionB'
optionB: does not have a value (no default or overridden with None - whatever), but resolve to: 'optionC'
optionC: has a value 'filesdir'

In this case, we want 'myfile' to be in 'filesdir'.

Here's the unit test for this usecase https://github.com/galaxyproject/galaxy/pull/8689/files#diff-6882d9765012ff6356c14d0e336c78c1R164